sysdig/resource_sysdig_secure_organization_test.go

```diff
@@ -0,0 +1,132 @@
+//go:build tf_acc_sysdig_secure || tf_acc_sysdig_common
+
+package sysdig_test
+
+import (
+	"bytes"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/draios/terraform-provider-sysdig/sysdig"
+)
+
+func TestAccSecureOrganization(t *testing.T) {
+	// XXX: TF acceptance tests for secure org onboarding need an actual existing gcp project
+	// along with an actual service_principal_key to scrape all folders and projects under the org.
+	// Without it POST /organizations call will fail with 500 error.
+	// Skipping the test based on this error when it occurs.
+	rText := func() string { return acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum) }
+	accID := rText()
+	organizationApiUrl := fmt.Sprintf(`%s/api/cloudauth/v1/organizations`, os.Getenv("SYSDIG_SECURE_URL"))
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			if v := os.Getenv("SYSDIG_SECURE_API_TOKEN"); v == "" {
+				t.Fatal("SYSDIG_SECURE_API_TOKEN must be set for acceptance tests")
+			}
+		},
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"sysdig": func() (*schema.Provider, error) {
+				return sysdig.Provider(), nil
+			},
+		},
+		ErrorCheck: func(err error) error {
+			// if regex matches with the expected error, do t.Skip
+			re := regexp.MustCompile(fmt.Sprintf(`POST %s giving up after 5 attempt(s)`, organizationApiUrl))
+			if re.MatchString(err.Error()) {
+				t.Skipf("skipping test; this POST call is not supported without actual existing GCP projects and service principal.")
+			}
+			return nil
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: secureOrgWithAccountID(accID),
+			},
+			{
+				ResourceName:            "sysdig_secure_cloud_auth_account.sample",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"component"},
+			},
+		},
+	})
+}
+
+func secureOrgWithAccountID(accountID string) string {
+	// this is a base64 encoded service account key
+	test_service_account_key_encoded := getEncodedGCPServiceAccountKeyForOrg("sample", accountID)
+
+	return fmt.Sprintf(`
+resource "sysdig_secure_cloud_auth_account" "sample" {
+  provider_id   = "%s"
+  provider_type = "PROVIDER_GCP"
+  enabled       = "true"
+  feature {
+	secure_config_posture {
+	  enabled    = "true"
+	  components = ["COMPONENT_SERVICE_PRINCIPAL/secure-posture"]
+	}
+	secure_identity_entitlement {
+	  enabled    = true
+	  components = ["COMPONENT_SERVICE_PRINCIPAL/secure-posture"]
+	}
+  }
+  component {
+	type                       = "COMPONENT_SERVICE_PRINCIPAL"
```

Comment:
nit: the spacing looks off compared to below L89-L96

Reply:
Interesting, it didn't show up in my IDE. Fixed it in next commit. Thanks :)